packages/wonder-blocks-dropdown/src/components/__tests__/single-select.test.js

```diff
@@ -1,111 +1,207 @@
+        const uncontrolledSingleSelect = (
+            <SingleSelect onChange={onChange} placeholder="Choose">
                 <OptionItem label="item 1" value="1" />
                 <OptionItem label="item 2" value="2" />
-            </SingleSelect>,
+                <OptionItem label="item 3" value="3" />
+            </SingleSelect>
         );
 
-        // Act
-        const opener = wrapper.find(SelectOpener).find("button");
+        describe("mouse", () => {
+            it("should open when clicking on the default opener", () => {
+                // Arrange
+                render(uncontrolledSingleSelect);
+                const opener = screen.getByText("Choose");
+
+                // Act
+                userEvent.click(opener);
+
+                // Assert
+                expect(screen.queryByRole("listbox")).toBeInTheDocument();
+            });
+
+            it("the opener should keep the focus after opening", () => {
+                // Arrange
+                mockRequestAnimationFrame(FRAME_DURATION);
+                render(uncontrolledSingleSelect);
+                const opener = screen.getByText("Choose");
 
-        // Assert
-        expect(opener.prop("data-test-id")).toBe("some-test-id");
+                // Act
+                userEvent.click(opener);
+                jest.advanceTimersByTime(FRAME_DURATION);
+
+                expect(screen.getByRole("button")).toHaveFocus();
+            });
+
+            it("should close when clicking on the default opener a second time", () => {
+                // Arrange
+                render(uncontrolledSingleSelect);
+                const opener = screen.getByText("Choose");
+                userEvent.click(opener);
+
+                // Act
+                userEvent.click(opener);
+
+                // Assert
+                expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+            });
+
+            it("should close when clicking on an item", () => {
+                // Arrange
+                render(uncontrolledSingleSelect);
+                const opener = screen.getByText("Choose");
+                userEvent.click(opener);
+
+                // Act
+                userEvent.click(screen.getByText("item 1"));
+
+                // Assert
+                expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+            });
+
+            it("should call onChange() with the item's value when clicking it", () => {
+                // Arrange
+                render(uncontrolledSingleSelect);
+                const opener = screen.getByText("Choose");
+                userEvent.click(opener);
+
+                // Act
+                userEvent.click(screen.getByText("item 1")); // closed
+
+                // Assert
+                expect(onChange).toHaveBeenCalledWith("1"); // value
+            });
+        });
+
+        describe("keyboard", () => {
+            describe.each([{key: "{enter}"}, {key: "{space}"}])(
```

Comment:
Nice improvement!

Reply:
The tests within this describe are testing more than we were testing before.